### PR TITLE
[eslint-plugin-packlets] Fix several miscellaneous issues

### PR DESCRIPTION
**Pull request by octogonz**
Fix a couple issues reported by @victor-wm:
- An exception occurred if a source file was added to the `src/packlets` folder, not belonging to any packlet
- linting was sometimes not performed on MacOS, because Node.js `path.relative()` incorrectly assumes that every POSIX file system is case-sensitive

While testing these fixes, I also noticed this problem and fixed it as well:
- `@rushstack/packlets/circular-deps` did not detect certain types of circular dependencies

### CODE REVIEW

File: stack/eslint-plugin-packlets/src/Path.ts [hunk 1]
```diff
@@ -2,9 +2,113 @@
...
 export class Path {
+  /**
+   * Whether the filesystem is assumed to be case sensitive for Path operations.
+   *
+   * @remarks
+   * Regardless of operating system, a given file system's paths may be case-sensitive or case-insensitive.
+   * If a volume is mounted under a subfolder, then different parts of a path can even have different
+   * case-sensitivity.  The Node.js "path" API naively assumes that all Windows paths are case-insensitive,
+   * and that all other OS's are case-sensitive.  This is way off, for example a modern MacBook has a
+   * case-insensitive filesystem by default.  There isn't an easy workaround because Node.js does not expose
```
Comment (octogonz) [inline stack/eslint-plugin-packlets/src/Path.ts:18]: @iclanton I'm amazed we didn't encounter this problem earlier.  It's very surprising.

Maybe we should consider deprecating `path` and apply this approach to `@rushstack/node-core-library`'s `Path` API.

File: stack/eslint-plugin-packlets/src/Path.ts [hunk 2]
```diff
@@ -2,9 +2,113 @@
 // See LICENSE in the project root for license information.
 
 import * as path from 'path';
```
Comment (iclanton) [inline stack/eslint-plugin-packlets/src/Path.ts:4]: Do we want to call this something like `nodeJsPath`?

File: stack/eslint-plugin-packlets/src/Path.ts [hunk 3]
```diff
@@ -2,9 +2,113 @@
...
+import * as fs from 'fs';
+
+export type ParsedPath = path.ParsedPath;
 
-// These helpers are borrowed from @rushstack/node-core-library
 export class Path {
```
Comment (iclanton) [inline stack/eslint-plugin-packlets/src/Path.ts:9]: Let's put together some tests for this.

File: stack/eslint-plugin-packlets/src/Path.ts (hunk 1 above)
Comment (iclanton) [reply stack/eslint-plugin-packlets/src/Path.ts:18]: I'm a little surprised as well. I'm cool with deprecating Node's `path` as long as our thing's perf isn't worse.

**Review by iclanton — APPROVED** [review]

File: stack/eslint-plugin-packlets/src/Path.ts (hunk 3 above)
Comment (octogonz) [reply stack/eslint-plugin-packlets/src/Path.ts:9]: Since the original code worked fine on Linux and Windows, but failed on Mac, there isn't an easy way to write tests for that. We would need a way to deterministically simulate API behavior for different OS's. Or else our CI would need to do builds on all 3 OS's.

I can try to write a test for the string manipulations in `_relativeCaseInsensitive()`.

Comment (octogonz) [reply stack/eslint-plugin-packlets/src/Path.ts:9]: Added some tests and fixed some obscure edge cases
